### PR TITLE
guid2english: fix short destination buffer #292

### PR DESCRIPTION
**Pull request by JulienVdG**
Processing the `src` at once is only valid if `src` contains the full text to
transform.
When the stream is longer, one need to process elements incrementally by
consuming only the beginning of the buffer if there is a risk that the end
contains only partial data.

Also processing the full `src` buffer at once will generate bigger buffer at
`dst` (most of the case the template adds stuff) so treating only the first
element will always fit in `dst`.

### CODE REVIEW

File: pkg/guid2english/transformer.go [hunk 1]
```diff
@@ -80,21 +84,51 @@ func New(m Mapper) *Transformer {
...
+			return d, s, transform.ErrShortDst
 		}
-		return t.mapper.Map(*g)
-	})
-	if len(transformed) > len(dst) {
-		return 0, 0, transform.ErrShortDst
+		copy(dst, transformed)
+		return len(transformed), len(src), nil
+	}
+	loc := guidRegex.FindIndex(src)
```
Comment (GanShun) [inline pkg/guid2english/transformer.go]: Doesn't this cause you to miss the partial guid if you have another GUID who's index can be found?

Comment (JulienVdG) [reply pkg/guid2english/transformer.go]: I don't get your point...
FindIndex will find the first match in src from the beginning of the buffer. a partial GUID can only be at the end, and we don't consume the end of the buffer so next call to Transform by the transform.Reader will give us the remains of src appended with new data (unless at the end of file where atEOF will eventually be set which cover the case of the file containing a partial GUID at its end)

Comment (rjoleary) [reply pkg/guid2english/transformer.go]: This looks correct to me. The return is `return loc[0] + len(mappedGUID), loc[1], transform.ErrShortSrc`, so only the first guid is consumed. A partial guid wold be left in the source buffer.

Comment (GanShun) [reply pkg/guid2english/transformer.go]: I think I misunderstood how the transform function is supposed to work, I think this is right

**Comment by rjoleary** [comment]
Thanks for fixing my crappy code!

**Comment by JulienVdG** [comment]
@rjoleary you're welcome, it was an opportunity to learn more of go... and to distract me while I'm supposed to write docs :rofl:

**Pull request merged**